pallets/nfts/src/features/approvals.rs

```diff
@@ -173,4 +170,102 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		Ok(())
 	}
+
+	pub(crate) fn do_approve_collection(
+		maybe_check_origin: Option<T::AccountId>,
+		collection: T::CollectionId,
+		delegate: T::AccountId,
+	) -> DispatchResult {
+		ensure!(
```

Comment:
These checks are necessarily duplicate with the checks in the `approve_transfer`